### PR TITLE
Creating header

### PR DESCRIPTION
**Pull request by f1nwe**
- [ ] @denys281 

- [ ] @Kmalynss 

- [ ] @igorkasyanchuk 

- Created header.
- Added temporary styles for it. (We will change it, when we implement new template, after first demo)
- Added image_url view in User model, for storing avatar from facebook.

link to issue: https://github.com/Kotany-IF070-WebUI-RoR/NewVasjuky/issues/26

### CODE REVIEW

File: app/views/layouts/_header.html.slim [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
-          span
-            | New
-          | Vasjuky
-        #menu-collapse
-          ul.user-menu.pull-right
-            li
-              button.btn.btn-primary
-                | Login with Facebook
-                / Place your facebook login code up here
+  .row
```
Comment (simplews) [inline app/views/layouts/_header.html.slim]: Are You sure about it?
I think we can use nav tag without row and container classes on top levels
See http://getbootstrap.com/components/#navbar and HTML source of http://getbootstrap.com/examples/starter-template/

File: app/assets/stylesheets/variables_and_mixins.scss [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+$color_st_tropaz: #285e8e;
+$black_25: rgba(0, 0, 0, 0.25);
+$color_black_haze: #f6f6f6;
+$color_alto: #ddd;
+$color_mercury: #e6e7e8;
+
+
+//Methods
+
+@mixin border-radius($radius) {
```
Comment (igorkasyanchuk) [inline app/assets/stylesheets/variables_and_mixins.scss]: remove this 
and use http://compass-style.org/reference/compass/css3/border_radius/
where you need

File: app/helpers/users_helper.rb [hunk 3]
```diff
@@ -1,3 +1,6 @@
 # frozen_string_literal: true
 module UsersHelper
+  def get_first_name(user)
```
Comment (igorkasyanchuk) [inline app/helpers/users_helper.rb]: remove this method
add "first_name" and "last_name" columns to User model
and take them from social networks auth.first_name (something like this) during registration

File: app/models/user.rb [hunk 4]
```diff
@@ -17,6 +17,7 @@ def self.from_omniauth(auth)
       user.email = auth.info.email
       user.password = Devise.friendly_token[0, 20]
       user.name = auth.info.name
+      user.image_url = auth.info.image
```
Comment (igorkasyanchuk) [inline app/models/user.rb]: user.first_name = auth.info.first_name (probably like this)

File: app/views/layouts/_header.html.slim [hunk 5]
```diff
@@ -1,20 +1,33 @@
...
+        .navbar-header
+          a.navbar-brand href="/"
+            span
+              | New
+            | Vasjuky
+          - if current_user.nil?
+            = link_to 'Sign in with Facebook', user_facebook_omniauth_authorize_path,
+                    class: ('btn btn-primary header-btn col-xs-12 col-sm-3 col-sm-offset-6 col-md-3 col-lg-2 col-lg-offset-7' ), \
+id:('facebook_auth')
+          - else current_user
```
Comment (igorkasyanchuk) [inline app/views/layouts/_header.html.slim]: else current_user? this is not ok
just else

File: app/views/layouts/_header.html.slim [hunk 6]
```diff
@@ -1,20 +1,33 @@
...
+            ul.navbar-nav.nav.pull-right
+              - if current_user.admin? || current_user.moderator?
```
Comment (igorkasyanchuk) [inline app/views/layouts/_header.html.slim]: we have already helper as I know `if admin_or_moderator?`

File: app/views/layouts/_header.html.slim [hunk 7]
```diff
@@ -1,20 +1,33 @@
...
+                  = link_to 'All users', account_users_path
+                li.nav-text-link
+                  = link_to 'Issues for approval', account_issues_path
+              li
+                = link_to 'Report issue', new_issue_path, class: 'btn btn-primary header-btn'
+
+              li.dropdown
+                a.dropdown-toggle href='#' data-toggle="dropdown"
+                  img.user_img src='#{current_user.image_url}'
+                  = get_first_name(current_user)
```
Comment (igorkasyanchuk) [inline app/views/layouts/_header.html.slim]: = current_user.first_name

File: spec/controllers/omniauth_callbacks_controller_spec.rb [hunk 8]
```diff
@@ -41,6 +41,8 @@ def env_for_omniauth
     OmniAuth::AuthHash.new(provider: 'facebook',
                            uid: '1234',
                            info: { name: 'Vasya',
-                                   email: 'vasya@pup.kin' })
+                                   email: 'vasya@pup.kin',
+                           image: 'http://www.bobbycrosby.com/pupkin/\
```
Comment (igorkasyanchuk) [inline spec/controllers/omniauth_callbacks_controller_spec.rb]: use image from local folder like File.open("#{Rails.root}/spec/files/photo.png")

File: app/helpers/users_helper.rb [hunk 9]
```diff
@@ -1,3 +1,10 @@
 # frozen_string_literal: true
 module UsersHelper
+  # def first_name(fullname)
```
Comment (igorkasyanchuk) [inline app/helpers/users_helper.rb]: remove this garbage

File: app/assets/stylesheets/variables_and_mixins.scss (hunk 2 above)
Comment (f1nwe) [inline app/assets/stylesheets/variables_and_mixins.scss]: fixed

File: app/views/layouts/_header.html.slim [hunk 10]
```diff
@@ -2,19 +2,13 @@ header
...
-          ul.user-menu.pull-right
-            li
-              button.btn.btn-primary
-                | Login with Facebook
-                / Place your facebook login code up here
+        - if current_user.nil?
+          = link_to 'Sign in with Facebook', user_facebook_omniauth_authorize_path,
+                  class: ('btn btn-primary header-btn col-xs-12 col-sm-3 col-sm-offset-6 col-md-3 col-lg-2 col-lg-offset-7' ), \
+id:('facebook_auth')
+        - else current_user
```
Comment (igorkasyanchuk) [inline app/views/layouts/_header.html.slim]:
```
if current_user
  ...
else # remove current_user after "else"
....
```

File: db/migrate/20161228201733_add_full_name_to_users.rb [hunk 11]
```diff
@@ -0,0 +1,6 @@
+class AddFullNameToUsers < ActiveRecord::Migration[5.0]
+  def change
+    add_column :users, :first_name, :string
```
Comment (igorkasyanchuk) [inline db/migrate/20161228201733_add_full_name_to_users.rb]: what about factory changes? 
we need to have validation for first_name, last_name and generate them in factory girl with faker

File: spec/controllers/omniauth_callbacks_controller_spec.rb [hunk 12]
```diff
@@ -40,7 +41,10 @@ def env_for_omniauth
   OmniAuth.config.mock_auth[:facebook] =
     OmniAuth::AuthHash.new(provider: 'facebook',
                            uid: '1234',
-                           info: { name: 'Vasya',
-                                   email: 'vasya@pup.kin' })
+                           info: { first_name: 'Vasya',
+                                   last_name: 'Pupkin',
+                                   email: 'vasya@pup.kin',
+                           image: "#{Rails.root}/spec/files/avatar.jpg"
```
Comment (igorkasyanchuk) [inline spec/controllers/omniauth_callbacks_controller_spec.rb]: File.open as I said before

File: app/helpers/users_helper.rb (hunk 9 above)
Comment (f1nwe) [inline app/helpers/users_helper.rb]: oh, i forgot about it

File: db/migrate/20161228201733_add_full_name_to_users.rb (hunk 11 above)
Comment (f1nwe) [inline db/migrate/20161228201733_add_full_name_to_users.rb]: fixed

**Pull request merged**